From abd1e0f2713bd97088a39e05d35e77c7ab228ec5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pasi=20Sj=C3=B6holm?= <pasi.sjoholm@jollamobile.com>
Date: Mon, 2 Mar 2015 10:09:33 +0200
Subject: [PATCH 13/22] gdhcp: Fix rapid commit option length check

Rapid commit option lenght is zero according to RFC3315, 22.14.

(cherry picked from commit 1b6fd8bf2604eac83d3ff6e831191a4767862667)
---
 gdhcp/client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdhcp/client.c b/gdhcp/client.c
index 99ec9e0..f54de79 100644
--- a/gdhcp/client.c
+++ b/gdhcp/client.c
@@ -2474,7 +2474,7 @@ static gboolean listener_event(GIOChannel *channel, GIOCondition condition,
 			rapid_commit = dhcpv6_get_option(packet6, pkt_len,
 							G_DHCPV6_RAPID_COMMIT,
 							&option_len, &count);
-			if (!rapid_commit || option_len == 0 ||
+			if (!rapid_commit || option_len != 0 ||
 								count != 1)
 				/* RFC 3315, 17.1.4 */
 				return TRUE;
-- 
1.9.1

